vtex/faststore
# Feat: Allows some section's component to be overridable

## What's the purpose of this pull request?

Allows some section's components to be overridable. For the first group of sections we have (Hero, Navbar, Breadcrumb, ProductShelf, BannerText, _ProductDetails_)

_ProductDetails is in another_ [PR](https://github.com/vtex/faststore/pull/1805)

1. **Hero Section**
     overridable components:
      - Hero: UIHero
      - HeroImage: UIHeroImage
      - HeroHeader: UIHeroHeader
      
2. **Breadcrumb Section**
    overridable components:
     - Breadcrumb: UIBreadcrumb
     - Icon: UIIcon
     
3. **Navbar Section**
    overridable components:
    - Navbar: UINavbar,
    - Logo,
    - SearchInput,
    - ButtonSignIn,
    - CartToggle,
    - NavbarLinks: UINavbarLinks,
    - NavbarLinksList: UINavbarLinksList,
    - RegionButton,
    - NavbarSlider: UINavbarSlider,
    - NavbarSliderHeader: UINavbarSliderHeader,
    - NavbarSliderContent: UINavbarSliderContent,
    - NavbarSliderFooter: UINavbarSliderFooter,

4. **BannerText**
     overridable components:
    - Banner
    - BannerContent

5. **ProductShelf**
     overridable components:
    - ProductShelf: UIProductShelf
    - __experimentalCarousel (Core's Carousel)
    - __experimentalProductCard (Core's ProductCard)


## How it works?
- We are defining a group of components of each section that we will allow overriding. When overriding a component you will replace the previous component with your new component. More details in the [documentation](https://evergreen.faststore.dev/docs/component-customization/override-component). 

## How to test it?
- To help with testing I left the overrides files with an example

For instance:
`core/src/customizations/components/overrides/Navbar.tsx`

```
const SECTION = 'Navbar' as const

const overrides: SectionOverride[typeof SECTION] = {
  name: SECTION,
  components: {
   // Navbar: { Component: CustomComponent } ,
    // Logo:  { Component: CustomComponent },
    // SearchInput:  { Component: CustomComponent },
    // ButtonSignIn:  { Component: CustomComponent },
    // CartToggle:  { Component: CustomComponent },
    // NavbarLinks:  { Component: CustomComponent },
  },
}
```
You can uncomment one of these lines and check if your component is being replaced. You can also create a new component and use it instead. 

### Starters Deploy Preview

[WIP]
